Content.Server/Weapons/Ranged/Systems/GunSystem.cs

```diff
@@ -132,6 +133,28 @@ private void OnBallisticPrice(EntityUid uid, BallisticAmmoProviderComponent comp
                 case CartridgeAmmoComponent cartridge:
                     if (!cartridge.Spent)
                     {
+                        if (gun.CompatibleAmmo != null &&
+                            !gun.CompatibleAmmo.Exists(ammoAllowed => ammoAllowed.Equals(cartridge.Prototype))
+                            && user != null)
+                        {
+                            var damage = new DamageSpecifier();
```

Comment:
either put this in the component yaml for flare gun, or do something *like* this seen in other parts of the codebase (forgor the exact syntax)
```cs
var damage = new DamageSpecifier()
{
    ("Blunt", 6)
};
```

Reply:
Done. In this way one could also implement it for other guns, but can also totally ignore it since it's nullable.